connectivity/check/features.go

```diff
@@ -488,28 +508,18 @@ func (ct *ConnectivityTest) ForceDisableFeature(feature Feature) {
 // isFeatureKNPEnabled checks if the Kubernetes Network Policy feature is enabled from the configuration.
 // Note that the flag appears in Cilium version 1.14, before that it was unable even thought KNPs were present.
 func (ct *ConnectivityTest) isFeatureKNPEnabled(namespace string, enableK8SNetworkPolicy bool) (bool, error) {
-	version, err := ct.client.GetRunningCiliumVersion(context.Background(), namespace)
```

Comment:
This doesn't seem to be the only occurance of `client.GetRunningCiliumVersion` within the repo - not even in the connectivity tests. Even though it might fixes the current issue - there's quite a chance of facing another error in other cases. It looks like a semantic version is expected.

An alternative approach would be to fix the problem "on the other side" - by actually using a semantic version as image tag in the Cilium Integration Tests in the Cilium Proxy repo - e.g. by using the content of the `VERSION` file as docker image tag in the cilium repository.

Reply:
> This doesn't seem to be the only occurance of `client.GetRunningCiliumVersion` within the repo - not even in the connectivity tests. Even though it might fixes the current issue - there's quite a chance of facing another error in other cases. It looks like a semantic version is expected.

I think it would be good to get rid of `GetRunningCiliumVersion` (i.e. deriving Cilium version from image tags) altogether and either use the version from the helm chart (if available, e.g. when installed from a release or when `--chart-directory` is used).

> An alternative approach would be to fix the problem "on the other side" - by actually using a semantic version as image tag in the Cilium Integration Tests in the Cilium Proxy repo - e.g. by using the content of the `VERSION` file as docker image tag in the cilium repository.
> 
> edit: ok it seems as the tag isn't a semantic version in other cases too - e.g. local kind deployment - where the tag is `local` 🤔

Yep, ideally we'd want it to work for development builds with an arbitrary image tag.

> example of running the latest cilium-cli from master against a locally deployed version
> 
> ```
> ❯ ./cilium connectivity test --test=l7
> connectivity test failed: unable to determine if KNP feature is enabled: failed to parse Cilium version -dev:local to semver
> ```

This was also reported as #1507 which I think this PR would fix.

I'd say let's merge this for now and then address `GetRunningCiliumVersion` in a follow-up PR.

Reply:
:100: agree